File: 0485-max-consecutive-ones/0485-max-consecutive-ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        maxCount = 0
        count = 0
        for n in nums:
            if (n == 1) : 
                count+=1
                maxCount = max(count, maxCount);
            else: count = 0;
        return maxCount     

File: 0485-max-consecutive-ones/0485-max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        stack<int> s;
        stack<int> empty;
        int ans  = 0;
        for(auto num : nums){
            if((s.empty() && num == 1) || (!s.empty() && num==1)) s.push(num);
            if(!s.empty() && num == 0 ){
                ans = max(ans,int(s.size()));
                s=empty;
            }
        }
        return max(ans,int(s.size()));
    }
};